framework/include/utils_nonunity/PerfGraph.h

```diff
@@ -373,8 +380,8 @@ class PerfGraph : protected ConsoleStreamInterface
   /// easier to sort
   std::vector<CumulativeSectionInfo *> _cumulative_section_info_ptrs;
 
-  /// Whether or not live printing is active
-  std::atomic<bool> _live_print_active;
```

Comment:
Hmm, I'm not sure removing the atomic here is a good idea. The live thread is active while we're setting it to not be, correct?

Reply:
Ah! This is misleading, I'm not actually _removing_ the atomic type, I'm just recovering a previously existing non-atomic variable by the name `_active` I removed in #27470 and then removing `_live_print_active` completely.

Reply:
What was the need for having both active and live_print_active in the first place?

Do we have sections of the code where we turn it off temporarily ?

Reply:
This would explain the need for an atomic

Reply:
> What was the need for having both active and live_print_active in the first place?
> 
> Do we have sections of the code where we turn it off temporarily ?

Exactly, that was what confused me in the first place as well.

As far as I understood, `_active` should control if the overarching `PerfGraph` functionality, which enables both the summary at the end and live printing, should be active.
`_live_print_active`, on the other hand, was intended to temporarily disable live printing, but giving the option to re-enable it at a later point if desired. We never used that functionality, so I removed it, by immediately terminating its thread.

Reply:
> This would explain the need for an atomic

Again, I'm not making a variable that was atomic, non-atomic.
I'm just completely removing the functionality that needed the atomic.

Reply:
> Again, I'm not making a variable that was atomic, non-atomic.
I'm just completely removing the functionality that needed the atomic.

yes I understand that. I was just rambling that that functionality that you are removing does need the atomic. 

So the motivation for removing the functionality is that it's simplifying the system? Or do you think it never worked as expected?

Reply:
That's right, we are never using the functionality to pause and later resume live printing. Once paused, we are just leaving it lingering, and as lightweight as it is, I just don't see the point of keeping that thread alive.